src/v2/ion/transformer.js

```diff
@@ -0,0 +1,185 @@
+import { _ } from 'okta';
+
+/**
+ * Transform the ion spec response into canonical format.
+ */
+
+ /**
+  * Authn V2 response
+  * @typedef {Object} AuthnResponse
+  */
+
+/**
+ * @typedef {Object} AuthnResult
+ * @property {CurrentState} currentState
+ * @property {Object=} user
+ * @property {Object=} factor
+ */
+
+/**
+ * @typedef {Object} CurrentState
+ * @property {string} stateHandle
+ * @property {string} step
+ * @property {string} intent
+ * @property {function} cancel
+ * @property {function} context
+ * @property {Remediation} remediation
+ */
+
+/**
+ * @typedef {Object} Remediation
+ * @property {RemediationObject[]} values
+ */
+
+/**
+ * @typedef {Object} RemediationObject
+ * @property {string} name
+ * @property {value[]=} value mandatory visible parameters
+ */
+
+const isObject = (x) => _.isObject(x);
+const isArray = (x) => Array.isArray(x);
+const containsAny = (xs, ys) => _.intersection(xs, ys).length > 0;
+
+const getRelObjectByName = (resp, relName) => {
+  const getRelBy_ = (result, resp_) => {
+    if (isObject(resp_)) {
+      if (resp_.rel && containsAny(resp_.rel, relName)) {
+        result.push(resp_);
+      } else {
+        Object.keys(resp_)
+          .forEach(k => {
+            getRelBy_(result, resp_[k]);
+          });
+      }
+    } else if (isArray(resp_)) {
+      resp_.forEach(x => getRelBy_(result, x));
+    }
+    return result;
+  };
+
+  return getRelBy_([], resp);
+};
+
+/**
+ *
+ * @returns {Object} actionObject
+ * @returns {string} actionObject.name
+ * @returns {function} actionObject.createRequest
+ */
+const createActionObj = (relObj) => {
+  const data = (relObj.value || [])
+    .filter(v => v.visible === false)
+    .map(v => { return { [v.name]: v.value }; })
+    .reduce((init, x) => Object.assign({}, init, x), {});
+
+  const needExtraData = relObj.value.length !== Object.keys(data).length;
+
+  const obj = {
+    name: relObj.name,
+    createRequest (extraData = {}) {
+      const reqData = needExtraData ? Object.assign({}, data, extraData) : data;
+      return {
+        method: relObj.method,
+        url: relObj.href,
+        data: reqData,
+      };
+    }
+  };
+
+  return obj;
+};
+
+/**
+ * @returns {Object.<string, function>} actions
+ */
+const createActions = (allCreateForms) => {
+  const allActions = allCreateForms.map(createActionObj);
+
+  // turns list of actions into a mapping
+  return allActions.reduce((init, action) => {
```

Comment:
Also here, we might be able to simplify L100-105 to:
```suggestion
  return allActions.reduce((init, action) => Object.assign(init, { [action.name]: action.createRequest }), {});
```

Reply:
same as last.